### PR TITLE
Adding buildscript for package drupal_libraries

### PR DESCRIPTION
**Pull request by vathsalashetty25**
## Checklist

- [x] Have you checked and followed all the points mention in the [CONTRIBUTING.MD](https://github.com/ppc64le/build-scripts/blob/master/CONTRIBUTING.md)
- [x] Did you run the script(s) on fresh container with `set -e` option enabled and observe success ?
- [ ] Did you have **Legal approvals** for patch files ?

### CODE REVIEW

File: d/drupal_libraries/drupal_libraries_ubi_8.5.sh [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  rm -rf $CORE_PACKAGE_NAME
+  echo "$CORE_PACKAGE_NAME  | $PACKAGE_VERSION | $OS_NAME | GitHub | Removed existing package if any"
+
+fi
+if ! git clone $CORE_PACKAGE_URL $CORE_PACKAGE_NAME; then
+        echo "------------------$PACKAGE_NAME:clone_fails---------------------------------------"
+                echo "$CORE_PACKAGE_URL $CORE_PACKAGE_NAME"
+        echo "$CORE_PACKAGE_NAME  |  $CORE_PACKAGE_URL |  $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  Clone_Fails"
+        exit 1
+fi
```
Comment (prashantkhoje) [inline d/drupal_libraries/drupal_libraries_ubi_8.5.sh:41]:
```suggestion
if [ -d "$CORE_PACKAGE_NAME" ] ; then
    rm -rf $CORE_PACKAGE_NAME
    echo "$CORE_PACKAGE_NAME  | $PACKAGE_VERSION | $OS_NAME | GitHub | Removed existing package"
fi
if ! git clone $CORE_PACKAGE_URL $CORE_PACKAGE_NAME; then
    echo "------------------$PACKAGE_NAME:clone_fails---------------------------------------"
    echo "$CORE_PACKAGE_URL $CORE_PACKAGE_NAME"
    echo "$CORE_PACKAGE_NAME  |  $CORE_PACKAGE_URL |  $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  Clone_Fails"
    exit 1
fi
```

File: d/drupal_libraries/drupal_libraries_ubi_8.5.sh [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  rm -rf $PACKAGE_NAME
+  echo "$PACKAGE_NAME  | $PACKAGE_VERSION | $OS_NAME | GitHub | Removed existing package if any"
+
+fi
+if ! git clone $PACKAGE_URL $PACKAGE_NAME; then
+        echo "------------------$PACKAGE_NAME:clone_fails---------------------------------------"
+                echo "$PACKAGE_URL $PACKAGE_NAME"
+        echo "$PACKAGE_NAME  |  $PACKAGE_URL |  $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  Clone_Fails"
+        exit 1
+fi
```
Comment (prashantkhoje) [inline d/drupal_libraries/drupal_libraries_ubi_8.5.sh:67]:
```suggestion
if ! composer install --no-interaction; then
    echo "------------------$PACKAGE_NAME:install_fails-------------------------------------"
    echo "$PACKAGE_URL $PACKAGE_NAME"
    echo "$PACKAGE_NAME  |  $PACKAGE_URL | $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  Install_Fails"
    exit 1
fi

composer require --dev phpunit/phpunit --with-all-dependencies ^7 --no-interaction

cd modules/
# Check if package exists
if [ -d "$PACKAGE_NAME" ] ; then
    rm -rf $PACKAGE_NAME
    echo "$PACKAGE_NAME  | $PACKAGE_VERSION | $OS_NAME | GitHub | Removed existing package"
fi
if ! git clone $PACKAGE_URL $PACKAGE_NAME; then
    echo "------------------$PACKAGE_NAME:clone_fails---------------------------------------"
    echo "$PACKAGE_URL $PACKAGE_NAME"
    echo "$PACKAGE_NAME  |  $PACKAGE_URL |  $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  Clone_Fails"
    exit 1
fi
```

File: d/drupal_libraries/drupal_libraries_ubi_8.5.sh [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        exit 1
+else
+        echo "------------------$PACKAGE_NAME:install_&_test_both_success-------------------------"
+        echo "$PACKAGE_URL $PACKAGE_NAME"
+        echo "$PACKAGE_NAME  |  $PACKAGE_URL | $PACKAGE_VERSION | $OS_NAME | GitHub  | Pass |  Both_Install_and_Test_Success"
+        exit 0
+fi
+
+# drupal-libraries has 3 types of test Unit/Functional/Kernel.Unit test dont need drupal framework and DB etc. So can be executed by given script.
+# Please follow README file for more information to run Functional and kernel test cases.
```
Comment (prashantkhoje) [inline d/drupal_libraries/drupal_libraries_ubi_8.5.sh:85]:
```suggestion
if ! ../vendor/phpunit/phpunit/phpunit ../modules/libraries/tests/src/Unit; then
    echo "------------------$PACKAGE_NAME:install_success_but_test_fails---------------------"
    echo "$PACKAGE_URL $PACKAGE_NAME"
    echo "$PACKAGE_NAME  |  $PACKAGE_URL | $PACKAGE_VERSION | $OS_NAME | GitHub | Fail |  Install_success_but_test_Fails"
    exit 1
else
    echo "------------------$PACKAGE_NAME:install_&_test_both_success-------------------------"
    echo "$PACKAGE_URL $PACKAGE_NAME"
    echo "$PACKAGE_NAME  |  $PACKAGE_URL | $PACKAGE_VERSION | $OS_NAME | GitHub  | Pass |  Both_Install_and_Test_Success"
    exit 0
fi

# Package drupal-libraries has 3 types of tests - Unit/Functional/Kernel. Unit tests don't need drupal framework and DB etc. So they can be executed by given script.
# Please follow README file for more information to run functional and kernel test cases.
```

**Comment by prashantkhoje** [comment]
Looks good to be merged.

File: d/drupal_libraries/README.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
+
+How to run drupal related modules test cases (Unit/Functional/Intergration).
+-------------
+
+Summary :-
+
+    To run test cases in drupal module we need build drupal core package and drupal complete framework which include one database,apache server,and drupal core package.
+    There are 3 type of tests in drupal unit,functional,intergration. For unit test we dont need drupal full framework like database.
+    Unit test does not use database testing as i tested some packages.
+
```
Comment (prashantkhoje) [inline d/drupal_libraries/README.md:10]:
```suggestion
    To run test cases in drupal module, we need drupal core package and drupal complete framework which includes a database, apache server and core package itself.
    There are 3 type of tests in drupal unit, functional, integration. For unit tests, we don't need drupal full framework like database.
    Unit tests don't use database.
```

File: d/drupal_libraries/README.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+*************************
+
+Copy following files From local into VM thats needed to run docker file succesfully.
+
```
Comment (prashantkhoje) [inline d/drupal_libraries/README.md:14]:
```suggestion
Copy following files into the VM that is needed to run docker file successfully.
```

File: d/drupal_libraries/README.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+    2) Dockerfile.drupal.ubi
+    3) drupal.zip
+    4) drupal_schema.sql
+
+Rename 2 below files:-
+
+    #cp Dockerfile.drupal.ubi Dockerfile
+    #cp automate_drupal.sh.txt automate_drupal.sh
+    #chmod +x automate_drupal.sh
+
```
Comment (prashantkhoje) [inline d/drupal_libraries/README.md:25]:
```suggestion
Rename following 2 files:-
    # cp Dockerfile.drupal.ubi Dockerfile
    # cp automate_drupal.sh.txt automate_drupal.sh
    # chmod +x automate_drupal.sh
```

File: d/drupal_libraries/README.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+Now we can create an image  using below command from dockerfile (Dockerfile.drupal.ubi i.e Dockerfile)
+
+     #docker build -t drupal_image .
+     #docker images
+
```
Comment (prashantkhoje) [inline d/drupal_libraries/README.md:31]:
```suggestion
Now create an image from dockerfile (Dockerfile.drupal.ubi i.e Dockerfile)
     # docker build -t drupal_image .
     # docker images
```

File: d/drupal_libraries/README.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+    su - postgres -c 'if [[ $(psql -l | grep dru2_pg) ]]; then     echo "Database already configured..."; else    /usr/pgsql-13/bin/pg_ctl -D /var/lib/pgsql/13/data/ start;     /usr/pgsql-13/bin/createdb -T template0 dru2_pg;     psql dru2_pg < /opt/app-root/src/drupal_schema.sql; fi'
+
+Step 2:-
+
+    /usr/sbin/httpd -k start;
+
+
+step 3 Go to :-
+
+    #cd /opt/app-root/src/drupal/modules
```
Comment (prashantkhoje) [inline d/drupal_libraries/README.md:55]:
```suggestion
Then run a container using that image.
    # docker run -it -d drupal_image /bin/bash
    # docker ps
    # docker exec -it a025d0a771a8 /bin/bash

Live drupal webserver will be available at http://<ip>:8081

After that we need to run below command inside the container. Run it in single command. Step 1 will take some time to exec.

Step 1:-
    su - postgres -c 'if [[ $(psql -l | grep dru2_pg) ]]; then     echo "Database already configured..."; else    /usr/pgsql-13/bin/pg_ctl -D /var/lib/pgsql/13/data/ start;     /usr/pgsql-13/bin/createdb -T template0 dru2_pg;     psql dru2_pg < /opt/app-root/src/drupal_schema.sql; fi'

Step 2:-
    /usr/sbin/httpd -k start;

step 3 Go to :-
    # cd /opt/app-root/src/drupal/modules
```

File: d/drupal_libraries/README.md [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+git clone  https://git.drupalcode.org/project/$PACKAGE_NAME
+cd $PACKAGE_NAME
+git checkout 8.x-3.0-beta1
+
+cd /opt/app-root/src/drupal/core/
+../vendor/bin/drush pm:enable $PACKAGE_NAME
+
+RUN TEST:---
+
+bash-4.4# ../vendor/phpunit/phpunit/phpunit ../modules/$PACKAGE_NAME/tests
```
Comment (prashantkhoje) [inline d/drupal_libraries/README.md:73]:
```suggestion
# ../vendor/phpunit/phpunit/phpunit ../modules/$PACKAGE_NAME/tests
```

File: d/drupal_libraries/README.md [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+Time: 58.19 seconds, Memory: 4.00 MB
+
+OK (25 tests, 48 assertions)
+
+Remaining deprecation notices (1)
+
+  1x: Drupal\Tests\BrowserTestBase::$defaultTheme is required in drupal:9.0.0 when using an install profile that does not set a default theme. See https://www.drupal.org/node/3083055, which includes recommendations on which theme to use.
+    1x in DefinitionDiscoveryFactoryTest::testDiscovery from Drupal\Tests\libraries\Functional\ExternalLibrary\Definition
+
+bash-4.4# ../vendor/phpunit/phpunit/phpunit ../modules/$PACKAGE_NAME/tests/src/Unit
```
Comment (prashantkhoje) [inline d/drupal_libraries/README.md:88]:
```suggestion
# ../vendor/phpunit/phpunit/phpunit ../modules/$PACKAGE_NAME/tests/src/Unit
```

File: d/drupal_libraries/README.md [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+PHPUnit 7.5.20 by Sebastian Bergmann and contributors.
+
+Testing ../modules/libraries/tests/src/Unit
+.......                                                             7 / 7 (100%)
+
+Time: 925 ms, Memory: 4.00 MB
+
+OK (7 tests, 7 assertions)
+bash-4.4# ../vendor/phpunit/phpunit/phpunit ../modules/$PACKAGE_NAME/tests/src/Functional
```
Comment (prashantkhoje) [inline d/drupal_libraries/README.md:97]:
```suggestion
# ../vendor/phpunit/phpunit/phpunit ../modules/$PACKAGE_NAME/tests/src/Functional
```

File: d/drupal_libraries/README.md [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+Time: 14.77 seconds, Memory: 4.00 MB
+
+OK (1 test, 10 assertions)
+
+Remaining deprecation notices (1)
+
+  1x: Drupal\Tests\BrowserTestBase::$defaultTheme is required in drupal:9.0.0 when using an install profile that does not set a default theme. See https://www.drupal.org/node/3083055, which includes recommendations on which theme to use.
+    1x in DefinitionDiscoveryFactoryTest::testDiscovery from Drupal\Tests\libraries\Functional\ExternalLibrary\Definition
+
+bash-4.4# ../vendor/phpunit/phpunit/phpunit ../modules/$PACKAGE_NAME/tests/src/Kernel
```
Comment (prashantkhoje) [inline d/drupal_libraries/README.md:112]:
```suggestion
# ../vendor/phpunit/phpunit/phpunit ../modules/$PACKAGE_NAME/tests/src/Kernel
```

**Review by prashantkhoje — CHANGES_REQUESTED** [review]

**Review by kandarpamalipeddi — APPROVED** [review]

**Pull request merged**